packages/x-charts/src/BarChart/BarPlot.tsx

```diff
@@ -238,7 +238,28 @@ const BarPlotRoot = styled('g', {
   overridesResolver: (_, styles) => styles.root,
 })({
   [`& .${barElementClasses.root}`]: {
-    transition: 'opacity 0.2s ease-in, fill 0.2s ease-in',
+    [`&.${barElementClasses.vertical}`]: {
+      transition: 'opacity 0.2s ease-in, fill 0.2s ease-in, height 0.5s ease-in, y 0.5s ease-in',
+
+      '@keyframes growHeight': {
+        from: { transform: 'scaleY(0%)' },
+        to: { transform: 'scaleY(100%)' },
+      },
```

Comment:
Scaling wont work for rounded corners.

Reply:
Makes sense. 

I need `animation` to grow from zero to the initial height.

I could use `@starting-style` ([docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@starting-style)), but I'm not sure if browser support is enough

Reply:
By "won't work" you mean that while animating rounded corners will become straight, right? 

Something like this: 

https://github.com/user-attachments/assets/fea80bbe-fcbf-4ee1-816f-1152040f953a

Reply:
yes